scripts/callbacks_generator/gen_callbacks.py

```diff
@@ -0,0 +1,1010 @@
+#!/usr/bin/env python3
+# Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import os
+import sys
+import copy
+import glob
+import yaml
+import logging
+import argparse
+from io import StringIO
+from typing import List, Mapping, Callable, Set
+
+log = logging.getLogger('gen_callbacks')
+
+PROXY_INSTANCES_COUNT = 5
+
+# Callbacks which have user_data parameter, but are not detected:
+CB_USER_DATA_OVERRIDES = {
+    'ime_language_requested_cb',
+    'ime_imdata_requested_cb',
+    'ime_geometry_requested_cb',
+    'image_util_decode_completed_cb',
+    'image_util_encode_completed_cb',
+}
+
+CB_PARAMETER_PATCHES: Mapping[str, Callable] = {
+    'sensor_accuracy_changed_cb':
+        lambda cb: cb.params[-1].set_name('user_data'),
+    'system_settings_iter_cb':
+        lambda cb: cb.params[-1].set_name('user_data'),
+    'sensor_events_cb':
+        lambda cb: cb.params[1].set_type('sensor_event_s*').set_name('events'),
+    'mv_barcode_detected_cb':
+        lambda cb: cb.params[3].set_type('const char**').set_name('messages'),
+}
+
+KNOWN_TYPES = {
+    'bool',
+    'bool*',
+    'char*',
+    'char**',
+    'double',
+    'float*',
+    'float**',
+    'int',
+    'int*',
+    'long',
+    'long long',
+    'unsigned',
+    'unsigned char',
+    'unsigned char*',
+    'unsigned char**',
+    'unsigned int',
+    'unsigned int*',
+    'unsigned long',
+    'unsigned long long',
+    'unsigned short',
+    'void',
+    'void*',
+    'void**',
+    'bool',
+    'bool*',
+    'char*',
+    'char**',
+    'double',
+    'float*',
+    'float**',
+    'int',
+    'int*',
+    'long',
+    'long long',
+    'size_t',
+    'time_t',
+    'unsigned',
+    'unsigned char',
+    'unsigned char*',
+    'unsigned char**',
+    'unsigned int',
+    'unsigned int*',
+    'unsigned long',
+    'unsigned long long',
+    'unsigned short',
+    'void',
+    'void*',
+    'void**',
+    'size_t',
+    'time_t',
+}
+
+SPECIAL_TYPES = {
+    'dnssd_browser_h': 'unsigned int',
+    'dnssd_service_h': 'unsigned int',
+    'mtp_device_h': 'int',
+    'mtp_object_h': 'int',
+    'mtp_storage_h': 'int',
+    'ssdp_browser_h': 'unsigned int',
+    'ssdp_service_h': 'unsigned int',
+    'result_set_cursor': 'void*',
+    'stte_result_time_cb': 'void*',
+    'stte_supported_language_cb': 'void*',
+    'ttse_supported_voice_cb': 'void*',
+    'vce_supported_language_cb': 'void*',
+    'Ecore_IMF_Input_Hints': 'some_enum',
+    'Ecore_IMF_Input_Panel_Lang': 'some_enum',
+    'Ecore_IMF_Input_Panel_Layout': 'some_enum',
+    'Ecore_IMF_Input_Panel_Return_Key_Type': 'some_enum',
+    'eom_output_id': 'unsigned int',
+    'location_coords_s': 'location_coords_s_copy',
+    'mv_rectangle_s': 'mv_rectangle_s_copy',
+    'wifi_direct_connection_state_cb_data_s': 'wifi_direct_connection_state_cb_data_s_copy',
+}
+
+EXTRA_HEADERS = {
+    '5.0': [
+    ],
+}
```

Comment:
This seems redundant and could be removed.

The comment in `scripts/README.md` should also be removed if this is removed.